docs/examples/TimePlotterExample.py

```diff
@@ -0,0 +1,117 @@
+from stonesoup.types.array import StateVector, CovarianceMatrix
+from stonesoup.types.state import GaussianState
+from stonesoup.updater.kalman import KalmanUpdater
+from stonesoup.plotter import TimeBasedPlotter
+
+# Models
+transition_model = CombinedLinearGaussianTransitionModel(
+    [ConstantVelocity(1), ConstantVelocity(1)])
+measurement_model = LinearGaussian(4, [0, 2], np.diag([0.5, 0.5]))
+
+start_time = datetime.datetime.now()
+
+
+# Simulators
+groundtruth_sim = MultiTargetGroundTruthSimulator(
+    transition_model=transition_model,
+    initial_state=GaussianState(
+        StateVector([[0], [0], [0], [0]]),
+        CovarianceMatrix(np.diag([1000, 10, 1000, 10])),
+        timestamp=start_time),
+    timestep=datetime.timedelta(seconds=5),
+    number_steps=200,
+    birth_rate=0.5,
+    death_probability=0.05
+)
+detection_sim = SimpleDetectionSimulator(
+    groundtruth=groundtruth_sim,
+    measurement_model=measurement_model,
+    meas_range=np.array([[-1, 1], [-1, 1]]) * 5000,  # Area to generate clutter
+    detection_probability=0.9,
+    clutter_rate=1,
+)
+
+# Filter
+predictor = KalmanPredictor(transition_model)
+updater = KalmanUpdater(measurement_model)
+
+# Data Associator
+hypothesiser = DistanceHypothesiser(predictor, updater, Mahalanobis(), missed_distance=3)
+data_associator = GNNWith2DAssignment(hypothesiser)
+
+# Initiator & Deleter
+deleter = CovarianceBasedDeleter(covar_trace_thresh=1E3)
+initiator = MultiMeasurementInitiator(
+    GaussianState(np.array([[0], [0], [0], [0]]),
+                  np.diag([0, 100, 0, 1000]),
+                  timestamp=start_time),
+    measurement_model=measurement_model,
+    deleter=deleter,
+    data_associator=data_associator,
+    updater=updater,
+    min_points=3,
+)
+
+# Tracker
+tracker = MultiTargetTracker(
+    initiator=initiator,
+    deleter=deleter,
+    detector=detection_sim,
+    data_associator=data_associator,
+    updater=updater,
+)
+
+for time, tracks in tracker.tracks_gen():
+    pass
+
+all_plotting = []
+colours = colors.cnames
+
+for idx, ground_truth in enumerate(groundtruth_sim.groundtruth_paths):
+    all_plotting.append(TimeBasedPlotter(plotting_data=ground_truth.states, legend_key='Ground Truth'+str(idx),
+                                         linestyle='--', marker='.', alpha=0.5,
+                                         color=random.choice(list(colours.values()))
+                                         ))
+
+for idx, track in enumerate(tracks):
+    all_plotting.append(TimeBasedPlotter(plotting_data=track.states, legend_key='Track'+str(idx),
+                                         linestyle='-', marker='x', alpha=0.5,
+                                         color=random.choice(list(colours.values()))
+                                         ))
+
+times_to_plot = [start_time + x * groundtruth_sim.timestep for x in range(groundtruth_sim.number_steps)]
+line_ani = TimeBasedPlotter.run_animation(times_to_plot, all_plotting)
```

Comment:
The animation looks good but doesn't seem to run properly for me - everything occurs in the last couple of seconds and the legend jumps around a bit. Not sure how to fix that myself...

Reply:
The MultiTargetGroundTruthSimulator appears to creating all the targets at the end of the simulation time. I don't know why.  I put the following at line 101. 
```
for idx, ground_truth in enumerate(groundtruth_sim.groundtruth_paths):
    print(min(state.timestamp for state in ground_truth.states)-start_time)
    print(max(state.timestamp for state in ground_truth.states) - start_time)
```
It only produces ground truths between 14 and 17 minutes after the simulation is suppose to start. 
This is the first time I've used the MultiTargetGroundTruthSimulator, so I'm not sure why it doesn't work. @jswright-dstl can you advise? 

The legend will jump around trying to get out of the way of the tracks